src/main/java/com/hazelcast/jdbc/JdbcDataBaseMetadata.java

```diff
@@ -779,7 +1231,15 @@ public boolean supportsBatchUpdates() {
 
     @Override
     public ResultSet getUDTs(String catalog, String schemaPattern, String typeNamePattern, int[] types) {
-        return JdbcResultSet.EMPTY;
+        return emptyResultSet(new SqlRowMetadata(asList(
+                new SqlColumnMetadata("TYPE_CAT", SqlColumnType.VARCHAR, true),
+                new SqlColumnMetadata("TYPE_SCHEM", SqlColumnType.VARCHAR, true),
+                new SqlColumnMetadata("TYPE_NAME", SqlColumnType.VARCHAR, false),
+                new SqlColumnMetadata("CLASS_NAME", SqlColumnType.VARCHAR, false),
+                new SqlColumnMetadata("DATA_TYPE", SqlColumnType.SMALLINT, false),
```

Comment:
You used SMALLINT instead of INTEGER:

```suggestion
                new SqlColumnMetadata("DATA_TYPE", SqlColumnType.INTEGER, false),
```
Also in many other places you used SMALLINT, doesn't look like an oversight. I checked JDK7 and JDK17 whether the type has changed, but it always was INT...